### ISSUE TITLE
get rid of the references to environment variables in config.go

### ISSUE THREAD

**Issue by jmazzitelli** [kiali/kiali#1837]
Back in the early days of the project, users were to hand-edit the ConfigMap to configure Kiali (now we use the operator and the Kiali CR). To avoid having the user enter EVERY possible setting in the ConfigMap, in `config.go` we hardcoded default values for each setting. If a user did not enter a value for a specific setteing in the ConfigMap, the value would obtain a hardcoded default value defined in config.go.

Because we didn't want to be stuck with hardcoded values for defaults (in case we needed to change it for, say, a hotfix and we didn't want to respin a new image) we provided a backdoor way of changing those hardcoded default values via environment variables.

But now things have changed. The Kiali CR is the user-facing config (not the ConfigMap), and the operator is the one that defines all the defaults (not the server in config.go). If the Kiali CR does not define a particular setting, the operator is responsible for setting it to some default value. In effect, this means the Kiali ConfigMap will have virtually every setting defined - thus the hardcoded defaults in config.go are rarely used and the environment variable overrides are essentially useless since the ConfigMap will have virtually all the values already set (I am hedging here with "virtually" because there may be one or a couple settings in the ConfigMap that the operator allows to be undefined (for special purposes that the Kiali server knows about - but these are rare, special cases and wouldn't have env var overrides and thus we can ignore those special cases).

We should rip out all of the code in config.go that refers to environment variables and hardcoded defaults and just assume the ConfigMap has everything defined. Or, at minimum, get rid of the backdoor environment variable lookups and just set the default ConfigMap structure with hardcoded defaults (if not just empty values). This will make the code leaner, easier to maintain, and easier to understand.

**Comment by jshaughn** [comment]
There may be associated doc to be updated on kiali.io.

**Comment by jmazzitelli** [comment]
What we could do is bring back env var overrides for certain settings. For example, the whole reason why I looked at this was due to my work in PR #1827 - I wanted the operator to set env vars (whose values are taken from secrets) and have those env vars override certain values in the ConfigMap. This will allow sensitive values such as passwords and tokens to be passed to the server via env vars taken from secrets rather than defined in the ConfigMap itself which could leak/expose those sensitive values.
